src/components/result-catalog/ResultCatalog.tsx

```diff
@@ -0,0 +1,151 @@
+import { FC, useEffect, useState } from 'react';
+import { fetchData } from '../../service/apiService';
+import { ApiInfo, IApi } from '../../utils/types/types';
+import ResultCard from '../result-card/ResultCard';
+import Pagination from '../pagination/Pagination';
+import {
+  Link,
+  Outlet,
+  Route,
+  Routes,
+  useNavigate,
+  useParams,
+} from 'react-router-dom';
+import '../components.css';
+import SideBar from '../sidebar/SideBar';
+import SelectItemPerPage from '../select/SelectItemPerPage';
+interface ResultCatalogProps {
+  queryParam: string;
+  startPage: boolean;
+  handleStopSearch: () => void;
+}
+
+const ResultCatalog: FC<ResultCatalogProps> = (props) => {
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [items, setItems] = useState<IApi[]>([]);
+  const [apiInfo, setApiInfo] = useState<ApiInfo>();
+  const [hasError, setHasError] = useState(false);
+  const { page } = useParams();
+  const [currentPage, setcurrentPage] = useState(Number(page));
+  const [isSideBarOpen, setIsSideBarOpen] = useState(false);
+  const [totalCard, setTotalCard] = useState(20);
+  const navigate = useNavigate();
+  const defaultApiCardPerPage = 20;
+  const startApiPage =
+    (Number(page) - 1) * (totalCard / defaultApiCardPerPage) + 1;
+
+  const totalPages = Math.ceil(totalCard / defaultApiCardPerPage);
+
+  useEffect(() => {
+    async function fetchDataForAllPages(
```

Comment:
It's a good practice to move this function out of useEffect hook and rewrite it to arrow function

Reply:
Thank's))